tests/test_data/test_shape.py

```diff
@@ -17,14 +17,243 @@
+
+    # returns None if not valid input
+    monkeypatch.setattr(Shape, "polygonFromGeoJson", lambda x: None)
+    shape = Shape.fromGeoJSON("")
+    assert shape is None
+
+
+@pytest.mark.data
+def testDataShape_Polygon(tmpConf, tmpDir, caplog):
+    """ Checks that geojson file is stored and Shape instance is created"""
+    dataPath = Path(tmpDir) / "data"
+    dataPath.mkdir(exist_ok=True)
+    tmpConf.dataPath = dataPath
+
+    fn = "simple_Fylker_0.json"
+    fn = Path("tests/test_data/data") / fn
+    with fn.open() as f:
+        data = json.load(f)
+        data = data["polygon"]
+
+    uuid = '33d0c48f-b58c-4b1a-b224-e93b03393cb3'
+    shape = Shape(uuid)
+
+    # create file ans see if it is returned:
+    path = dataPath / f'{uuid}.geojson'
+    with path.open('w') as f:
+        json.dump(data, f)
+    result = shape.polygon()
+    assert isinstance(result, (Polygon, MultiPolygon))
+
+    # error if it does exist (default)
+    caplog.clear()
+    tolerance = 50.5
+    tolerance_str = str(50_500_000)
+    fnjson = (dataPath / f'{uuid}.{tolerance_str}.geojson')
+    # TODO: replace by saveunlink (missing_ok not available in py3.7)
+    if fnjson.exists():
+        fnjson.unlink()
+    result = shape.polygon(tolerance=tolerance)
+    assert "Polygon does not exist for tolerance" in caplog.text
+
+    # but create it, if cachedOnly=False
+    caplog.clear()
+    if fnjson.exists():
+        fnjson.unlink()
+    result = shape.polygon(tolerance=50.5, cachedOnly=False)
+    assert "Creating polygon" in caplog.text
+    assert fnjson.exists()
+
+    # now it exists -- check that it is loaded
+    shape.polygonFromGeoJson = lambda path: path
+    result = shape.polygon(tolerance=50.5, cachedOnly=False)
+    assert result == fnjson
+
+
+@pytest.mark.parametrize("fn",
+    ["simple_Fylker_0.json", "simple_Kommuner_0.json", "simple_Kommuner_291.json"]) # noqa
+@pytest.mark.data
+def testDataShape_ToGeoJson(fn, tmpConf, tmpDir):
+    """ Checks that geojson file is stored and Shape instance is created"""
+    dataPath = Path(tmpDir) / "data"
+    dataPath.mkdir(exist_ok=True)
+    tmpConf.dataPath = dataPath
+
+    fn = Path("tests/test_data/data") / fn
+    with fn.open() as f:
+        data = json.load(f)
+        data = data["polygon"]
+
+    shape = Shape.fromGeoJSON(data)
+    result = shape.toGeoJson()
+    expected = {"type": "Feature", "geometry": data}
+    # TODO:
+    # hacky comparison, but convert tuples to lists
+    result = eval(result.__repr__().replace("(", "[").replace(")", "]"))
+    assert result == expected
+
+
+@pytest.mark.parametrize("file, typ", [("simple_Fylker_0.json", Polygon),
+                                       ("simple_Kommuner_0.json", Polygon),
+                                       ("simple_Kommuner_291.json", MultiPolygon)])
+@pytest.mark.data
+def testDataShape_PolygonFromGeoJson(file, typ, tmpConf, tmpDir, caplog):
+    """ Checks that geojson file is stored and Shape instance is created"""
```

Comment:
Remove redundant space at start of docstring. (There are multiple cases of this)